srcs/juloo.keyboard2/Keyboard2.java

```diff
@@ -288,6 +286,64 @@ public void setInputView(View v)
     if (parent != null && parent instanceof ViewGroup)
       ((ViewGroup)parent).removeView(v);
     super.setInputView(v);
+    updateSoftInputWindowLayoutParams();
+  }
+
+
+  @Override
+  public void updateFullscreenMode() {
+    super.updateFullscreenMode();
+    updateSoftInputWindowLayoutParams();
+  }
+
+  private void updateSoftInputWindowLayoutParams() {
+    final Window window = getWindow().getWindow();
+    updateLayoutHeightOf(window, ViewGroup.LayoutParams.MATCH_PARENT);
+    final View inputArea = window.findViewById(android.R.id.inputArea);
+
+    updateLayoutHeightOf(
+            (View) inputArea.getParent(),
+            isFullscreenMode()
+                    ? ViewGroup.LayoutParams.MATCH_PARENT
+                    : ViewGroup.LayoutParams.WRAP_CONTENT);
+    updateLayoutGravityOf((View) inputArea.getParent(), Gravity.BOTTOM);
+
+  }
+
+  private static void updateLayoutHeightOf(final Window window, final int layoutHeight) {
+    final WindowManager.LayoutParams params = window.getAttributes();
+    if (params != null && params.height != layoutHeight) {
+      params.height = layoutHeight;
+      window.setAttributes(params);
+    }
+  }
+
+  private static void updateLayoutHeightOf(final View view, final int layoutHeight) {
+    final ViewGroup.LayoutParams params = view.getLayoutParams();
+    if (params != null && params.height != layoutHeight) {
+      params.height = layoutHeight;
+      view.setLayoutParams(params);
+    }
+  }
+
+  private static void updateLayoutGravityOf(final View view, final int layoutGravity) {
+    final ViewGroup.LayoutParams lp = view.getLayoutParams();
+    if (lp instanceof LinearLayout.LayoutParams) {
+      final LinearLayout.LayoutParams params = (LinearLayout.LayoutParams) lp;
+      if (params.gravity != layoutGravity) {
+        params.gravity = layoutGravity;
+        view.setLayoutParams(params);
+      }
+    } else if (lp instanceof FrameLayout.LayoutParams) {
+      final FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) lp;
+      if (params.gravity != layoutGravity) {
+        params.gravity = layoutGravity;
+        view.setLayoutParams(params);
+      }
+    } else {
+      throw new IllegalArgumentException(
+              "Layout parameter doesn't have gravity: " + lp.getClass().getName());
```

Comment:
This `throw` is not needed. The layout gravity does not need to be set in most cases and potentially crashing the app is not acceptable.